trieste/acquisition/sampler.py

```diff
@@ -194,330 +172,48 @@ def binary_search(val: float) -> float:  # Find empirical interquartile range
-            shape `[..., B, D]`, for batches of size `B` of points of dimension `D`. Must have a
-            consistent batch size across all calls to :meth:`sample` for any given
-            :class:`BatchReparametrizationSampler`.
-        :param jitter: The size of the jitter to use when stabilising the Cholesky decomposition of
-            the covariance matrix.
-        :return: The samples, of shape `[..., S, B, L]`, where `S` is the `sample_size`, `B` the
-            number of points per batch, and `L` the dimension of the model's predictive
-            distribution.
-        :raise ValueError (or InvalidArgumentError): If any of the following are true:
-            - ``at`` is a scalar.
-            - The batch size `B` of ``at`` is not positive.
-            - The batch size `B` of ``at`` differs from that of previous calls.
-            - ``jitter`` is negative.
-        """
-        tf.debugging.assert_rank_at_least(at, 2)
-        tf.debugging.assert_greater_equal(jitter, 0.0)
-
-        batch_size = at.shape[-2]
-
-        tf.debugging.assert_positive(batch_size)
-
-        if self._initialized:
-            tf.debugging.assert_equal(
-                batch_size,
-                tf.shape(self._eps)[-2],
-                f"{type(self).__name__} requires a fixed batch size. Got batch size {batch_size}"
-                f" but previous batch size was {tf.shape(self._eps)[-2]}.",
-            )
-
-        mean, cov = self._model.predict_joint(at)  # [..., B, L], [..., L, B, B]
-
-        if not self._initialized:
-            self._eps.assign(
-                tf.random.normal(
-                    [tf.shape(mean)[-1], batch_size, self._sample_size], dtype=tf.float64
-                )  # [L, B, S]
-            )
-            self._initialized.assign(True)
-
-        identity = tf.eye(batch_size, dtype=cov.dtype)  # [B, B]
-        cov_cholesky = tf.linalg.cholesky(cov + jitter * identity)  # [..., L, B, B]
-
-        variance_contribution = cov_cholesky @ tf.cast(self._eps, cov.dtype)  # [..., L, B, S]
-
-        leading_indices = tf.range(tf.rank(variance_contribution) - 3)
-        absolute_trailing_indices = [-1, -2, -3] + tf.rank(variance_contribution)
-        new_order = tf.concat([leading_indices, absolute_trailing_indices], axis=0)
-
-        return mean[..., None, :, :] + tf.transpose(variance_contribution, new_order)
-
-
-TrajectoryFunction = Callable[[TensorType], TensorType]
-"""
-Type alias for trajectory functions.
-
-An :const:`TrajectoryFunction` evaluates a particular sample at a set of `N` query
-points (each of dimension `D`) i.e. takes input of shape `[N, D]` and returns
-shape `[N, 1]`.
-
-A key property of these trajectory functions is that the same sample draw is evaluated
-for all queries. This property is known as consistency.
-"""
-
-
-class RandomFourierFeatureThompsonSampler(ThompsonSampler):
+class ThompsonSamplerFromTrajectory(ThompsonSampler):
```

Comment:
I'm a bit confused by the name - why do we call these samples trajectories?

Reply:
Trajectories is a well used term in the literature referring to sample paths from the GP. These are smooth cts function draws that can be optimized (using gradients) to find their optima, e.g. for Thompson sampling.

Reply:
Note that these are distinct to what is provided by a reparam sampler, which just provides (repeatable) samples at a fixed set of points.

Reply:
I don't see why people would call them a trajectory, besides that they look like that visually, we can be more precise than that e.g. calling them function draws 

but fair enough, if its a widespread name, sounds good

Reply:
Trajectory or path are equally good I think.